tests/test_gen_artifacts.py

```diff
@@ -0,0 +1,101 @@
+import click.testing
+
+from exhibitor_tls_artifacts.gen_artifacts import app
+
+
+class TestCLI:
+
+    @staticmethod
+    def _validate_files(directory):
+        """ Ensures that artifacts are created and that they have the
+        appropriate permission """
+        files = [('client-cert.pem', 0o100644), ('client-key.pem', 0o100600),
+                 ('clientstore.jks', 0o100600), ('root-cert.pem', 0o100644),
+                 ('serverstore.jks', 0o100600), ('truststore.jks', 0o100600)]
+        for f, mode in files:
+            full_path = directory / f
+            assert full_path.exists()
+            assert full_path.stat()[stat.ST_MODE] == mode
+
+    def test_default(self):
+        """ Tests the most basic operation """
+        runner = click.testing.CliRunner()
+
+        with runner.isolated_filesystem() as temp:
+            temp_path = Path(temp)
+            result = runner.invoke(app,
+                                   args=['10.10.10.10'],
+                                   catch_exceptions=False)
+
+            assert result.exit_code == 0
+            artifact_path = temp_path / 'artifacts' / 'node_10_10_10_10'
+            assert artifact_path.exists()
+            self._validate_files(artifact_path)
+
+    def test_custom_dir(self, tmp_path):
+        """ Test outputting artifacts to a customer location """
+        runner = click.testing.CliRunner()
+        new_path = tmp_path / 'new'
+        result = runner.invoke(app,
+                               args=['-d', new_path, '10.10.10.10'],
+                               catch_exceptions=False)
+
+        assert result.exit_code == 0
+        self._validate_files(new_path / 'node_10_10_10_10')
+
+    def test_dir_exists(self):
+        """ Test error case when the output directory already exists """
+        runner = click.testing.CliRunner()
+
+        with runner.isolated_filesystem() as temp:
+            err = textwrap.dedent("""\
+            Usage: exhibitor-tls-artifacts [OPTIONS] [NODES]...
+
+            Error: Invalid value for "-d" / "--output-directory": Artifacts directory cannot exist!
+            """)
+            temp_path = Path(temp)
+            output_dir = temp_path / 'exists'
+            output_dir.mkdir()
+            result = runner.invoke(app,
+                                   args=['-d', output_dir, '10.10.10.10'],
+                                   catch_exceptions=False)
+            assert result.exit_code == 2
+            assert err == result.stdout
+
+    def test_no_nodes(self):
+        """ Test error case if no arguments are provided """
+        err = textwrap.dedent("""\
+        Usage: exhibitor-tls-artifacts [OPTIONS] [NODES]...
+
+        Error: Invalid value for "-d" / "--output-directory": Artifacts directory cannot exist!
+        """)
+        runner = click.testing.CliRunner()
+        result = runner.invoke(app, catch_exceptions=False)
+        assert result.exit_code == 2
+        print(result.stdout)
```

Comment:
Should we use `print` here? Maybe remove or convert to `log.debug`?

Reply:
debugging print statement, I'll remove it :)